### ISSUE TITLE
Official Docker image?

### ISSUE THREAD

**Issue by sammcj** [snipe/snipe-it#437]
I was wondering if you were looking to make an official Docker image / dockerfile for SnipeIT?

It would certainly make deployment and upgrades a lot easier.

**Comment by snipe** [comment]
We don't currently have one, but would love it if someone made one.

**Comment by uberbrady** [comment]
For anyone interested in this - I have the beginnings of a Dockerfile setup for SnipeIT.

The thing I'm wondering about is whether or not I ought to build in the MySQL DB part. It'd be much easier if we had a published image/Dockerfile that built-in the MySQL part - someone could have this up and running in no time.

But the "docker way" is to have separated docker resources for each service, and then hook them together using ```docker link```.

I'm leaning towards the latter way, plus a README of some sort explaining how to walk through the various steps to get up-and-running.

Anyone who's interested in this feature, please comment here and let me know what it is that you're looking for.

**Comment by sammcj** [comment]
@uberbrady great! - I would suggest just making sure you have the requirements in your Dockerfile to support connecting to an external database (in docker or otherwise) via environment variables.

Out of interested does it *have* to be MySQL? - With most people moving to PostgreSQL for it's performance and reliability it would advantageous to add connectors (clients) for both MySQL and PostgreSQL into the Dockerfile.

**Comment by snipe** [comment]
It doesn't have to be MySQL, however there are some migrations that use SQL commands (like MODIFY) that don't play well with all SQL databases. We've run into this issue with SQL Server before. I assume it would work fine on PgSQL, but I haven't personally tested it.

**Comment by uberbrady** [comment]
Once we get something up and running that people can actually use, we can look into figuring out how to connect things up to Postgres, but let's at least get something running first.

**Comment by snipe** [comment]
+1 @uberbrady

**Comment by uberbrady** [comment]
Ugh. This is hard.

I have one that works, but I think mail doesn't work yet, and I still have quite a bit of testing to do. But here's the start:

https://github.com/uberbrady/snipe-it/tree/dockerized

I'll keep it updated with origin/master and hopefully have enough for a pull request soon.

**Comment by snipe** [comment]
![1237811519_chuck-norris-approves](https://cloud.githubusercontent.com/assets/197404/6223427/f9a4dbc6-b61d-11e4-972f-40d03be0499b.gif)

**Comment by snipe** [comment]
(But please use `develop` as your updating branch, per the [Contribution Guidelines](https://github.com/snipe/snipe-it/blob/develop/CONTRIBUTING.md) ;)

It honestly shouldn't matter tho, since your docker stuff shouldn't shit on app files.

**Comment by uberbrady** [comment]
I tried to clean up my funny branch but it's somehow unalterably smooshed up with some master stuff - or something. I'm not even sure.

I just made a brand-new branch based off of develop using ```git cherry-pick``` - (which is the first time I've ever had to use that command).

The resulting branch - https://github.com/uberbrady/snipe-it/tree/docker2 - looks like it will apply much more cleanly to develop.
